app/views/posts/show.html.erb

```diff
@@ -1,18 +1,23 @@
 <% content_for :title, "#{@user.name} | #{@post.title}" %>
+
 <div class="mainContainer">
-<%= render "user_post", post: @post%>
-<p> <%= @user.posts_counters%> </p>
-<div class="commentsDiv">
-  <% @comments.each do |comment| %>
-    <div >
-      <p>
-        <span><strong><%= comment.user.name %>:</strong></span>
-        <span><%= comment.text %></span>
-      </p>
-    </div>
-  <% end %>
+  <%= render "user_post", post: @post%>
+  <p> <%= @user.posts_counters%> </p>
+  <div class="commentsDiv">
+    <% @comments.each do |comment| %>
+      <div>
+        <p>
+          <span><strong><%= comment.user.name %>:</strong></span>
+          <span><%= comment.text %></span>
+          <% if current_user && comment.user == current_user %>
+            <%= link_to "Delete", user_post_comment_path(@user, @post, comment), method: :delete, 
+            data: { confirm: "Are you sure you want to delete this comment?" } %>
+          <% end %>
```

Comment:
- [ ] Great work creating the delete button. However, I noticed that it doesn't work for two reasons:

1. It makes `GET` requests when clicked instead of `DELETE` requests. Please see the screenshot below:

![image](https://github.com/Fahim-Khan-P/blog/assets/77106962/36963f6c-33d3-4914-a19b-17687cce3136)

2. The action for destroying a comment in the **_Comments Controller_** is a _**private**_ method and it would not be possible to delete comments because private methods can only be called within a class in Ruby. 😎 👍 

#### Hint: I would advise you to:
1. Use a `button_to` helper instead of the `link_to` helper in the `app/views/posts/show.html.erb` file 💻 
2. Remove the `button_to` helper that you created for deleting a comment in the `app/views/posts/index.html.erb` because it leads to the `Post#show` page due to the `link_to` helper that it is wrapped in. 👍 
3. Make the `destroy` action in the _**Comments Controller**_ a public method. 🚗 🍒